packages/generator/ni_measurement_plugin_sdk_generator/client/templates/measurement_plugin_client.py.mako

```diff
@@ -111,17 +115,43 @@ class ${class_name}:
         self._pin_map_client = pin_map_client
         self._stub: Optional[v2_measurement_service_pb2_grpc.MeasurementServiceStub] = None
         self._measure_response: Optional[
-            Generator[v2_measurement_service_pb2.MeasureResponse, None, None]
+            grpc.CallIterator[v2_measurement_service_pb2.MeasureResponse]
         ] = None
-        self._configuration_metadata = ${configuration_metadata}
-        self._output_metadata = ${output_metadata}
+        self._configuration_metadata = {
+            % for key, value in configuration_metadata.items():
+            ${key}: ParameterMetadata(
+                display_name="${value.display_name}",
+                type=Field.Kind.ValueType(${value.type}),
+                repeated=${value.repeated},
+                default_value=${_format_default_value(value.default_value)},
+                annotations=${value.annotations},
+                message_type="${value.message_type}",
+                field_name="${value.field_name}",
+                enum_type=${_replace_enum_class_type(str(value.enum_type))}
```

Comment:
This is https://github.com/ni/measurement-plugin-python/pull/901#discussion_r1766965181 again.

Now you have the actual type object, so remove the regex hack.

Reply:
Sure, I removed the helper method with a regex approach and updated the code to use `value.enum_type.__name__` to extract the enum type from the type object (for example, `<enum 'EnumInEnum'>`).